sonar-batch/src/main/java/org/sonar/batch/ProjectConfigurator.java

```diff
@@ -94,11 +95,11 @@ private void checkCurrentAnalysisIsTheLatestOne(String projectKey, Date analysis
       ResourceModel persistedProject = databaseSession.getSingleResult(ResourceModel.class, "key", projectKey, "enabled", true);
       if (persistedProject != null) {
         Snapshot lastSnapshot = databaseSession.getSingleResult(Snapshot.class, "resourceId", persistedProject.getId(), "last", true);
-        if (lastSnapshot != null && !lastSnapshot.getCreatedAt().before(analysisDate)) {
+        if (lastSnapshot != null && !timeToDate(lastSnapshot.getCreatedAt()).before(analysisDate)) {
```

Comment:
instead of building a new Date object, you can compare longs:
lastSnapshot.getCreatedAt()>=analysisDate.getTime()